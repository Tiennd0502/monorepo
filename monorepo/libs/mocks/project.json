{
  "name": "mocks",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/mocks/src",
  "projectType": "library",
  "tags": [],
  "// targets": "to see all targets run: nx show project mocks --web",
  "targets": {}
}
