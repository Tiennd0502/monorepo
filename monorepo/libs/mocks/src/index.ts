export * from './blog';
export * from './user';
export * from './orders';
export * from './product';
export * from './category';
