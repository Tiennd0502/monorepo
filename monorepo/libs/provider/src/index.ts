export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
