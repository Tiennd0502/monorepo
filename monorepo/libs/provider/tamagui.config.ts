import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
