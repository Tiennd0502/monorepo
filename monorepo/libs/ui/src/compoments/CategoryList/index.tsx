import { memo, useState } from 'react';

// Types
import { Category } from '@monorepo/types';

// Components
import { ScrollView } from 'tamagui';
import CategoryItem from '../CategoryItem';

interface CategoryListProps {
  list: Category[];
  onChange: (value: string) => void;
}

const CategoryList = ({ list, onChange }: CategoryListProps) => {
  const [activeId, setActiveId] = useState(list[0].id);

  return (
    <ScrollView
      horizontal
      contentContainerStyle={{
        columnGap: 20,
      }}
      showsHorizontalScrollIndicator={false}
    >
      {list.map((item, index) => {
        const handlePress = () => {
          setActiveId(item.id);
          item.id !== activeId && onChange(item.id);
        };
        return (
          <CategoryItem
            key={item.label + index}
            isActive={activeId === item.id}
            item={item}
            onPress={handlePress}
          />
        );
      })}
    </ScrollView>
  );
};

export default memo(CategoryList);
