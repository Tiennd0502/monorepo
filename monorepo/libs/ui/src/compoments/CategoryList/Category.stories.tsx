import { Meta, StoryObj } from '@storybook/react';

import { CATEGORIES } from '@monorepo/mocks';

import CategoryList from '.';

const meta: Meta<typeof CategoryList> = {
  title: 'Components/CategoryList',
  component: CategoryList,
};

export default meta;

type Story = StoryObj<typeof CategoryList>;

export const Default: Story = {
  args: {
    list: CATEGORIES,
    onChange: () => null,
  },
};
