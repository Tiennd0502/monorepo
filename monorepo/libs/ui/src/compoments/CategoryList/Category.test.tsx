import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import {CATEGORIES} from '@monorepo/mocks';
import CategoryList from '.';

const onChange = jest.fn();

const baseProps = {
  list: CATEGORIES,
  onChange,
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <CategoryList {...props} />
    </TamaguiProvider>
  );
};

describe('CategoryList', () => {
  it('should render CategoryList component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });
});
