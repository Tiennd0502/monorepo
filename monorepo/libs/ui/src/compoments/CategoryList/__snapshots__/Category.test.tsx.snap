// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render CategoryList component 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "columnGap": 20,
    }
  }
  horizontal={true}
  scrollEnabled={true}
  showsHorizontalScrollIndicator={false}
>
  <View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "gap": 8,
        }
      }
      testID="category"
    >
      <View
        accessibilityRole="button"
        focusVisibleStyle={{}}
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#242424",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": "fit-content",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          alt="Polular"
          source={
            {
              "uri": "https://i.ibb.co/PQztKcN/star-1.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#242424",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Polular
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "gap": 8,
        }
      }
      testID="category"
    >
      <View
        accessibilityRole="button"
        focusVisibleStyle={{}}
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#909090",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": "fit-content",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          alt="Chair"
          source={
            {
              "uri": "https://i.ibb.co/g4xqj2d/chair-4.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#999",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Chair
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "gap": 8,
        }
      }
      testID="category"
    >
      <View
        accessibilityRole="button"
        focusVisibleStyle={{}}
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#909090",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": "fit-content",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          alt="Table"
          source={
            {
              "uri": "https://i.ibb.co/wRXzDHj/table-1-1.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#999",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Table
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "gap": 8,
        }
      }
      testID="category"
    >
      <View
        accessibilityRole="button"
        focusVisibleStyle={{}}
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#909090",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": "fit-content",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          alt="ArmChair"
          source={
            {
              "uri": "https://i.ibb.co/qpdkgf8/sofa-1.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#999",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        ArmChair
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "gap": 8,
        }
      }
      testID="category"
    >
      <View
        accessibilityRole="button"
        focusVisibleStyle={{}}
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#909090",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": "fit-content",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          alt="Bed"
          source={
            {
              "uri": "https://i.ibb.co/d2XdhGp/bed-1.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#999",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Bed
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "center",
          "alignItems": "center",
          "gap": 8,
        }
      }
      testID="category"
    >
      <View
        accessibilityRole="button"
        focusVisibleStyle={{}}
        focusable={true}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#909090",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": "fit-content",
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <Image
          alt="Lamb"
          source={
            {
              "uri": "https://i.ibb.co/sw7RkYc/lamp-1.png",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#999",
            "fontFamily": "NunitoSans_7pt-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          }
        }
        suppressHighlighting={true}
      >
        Lamb
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
